devito/__init__.py

```diff
@@ -10,7 +10,8 @@
 from devito.dimension import *  # noqa
 from devito.equation import *  # noqa
 from devito.finite_difference import *  # noqa
-from devito.function import Buffer  # noqa
+from devito.function import (PrecomputedSparseFunction, # noqa
+                             PrecomputedSparseTimeFunction, Buffer) # noqa
```

Comment:
this is not gonna be supported by YASK. We should never import from ``devito.function``, but rather from ``devito`` -- the backend mechanism does the rest. 

The new Function types must rather be added here: https://github.com/opesci/devito/blob/master/devito/base.py

Reply:
At this point, I'm curious what purpose this serves: https://github.com/opesci/devito/blob/master/devito/function.py#L23

Reply:
perhaps they're unnecessary